packages/lodestar/src/api/impl/validator/validator.ts

```diff
@@ -139,14 +143,24 @@ export class ValidatorApi implements IValidatorApi {
       return  state.validators.findIndex((v) => this.config.types.BLSPubkey.equals(v.pubkey, publicKey));
     }));
 
-    return validatorIndexes.map((validatorIndex) => {
+    const attesterDuties = validatorIndexes.map((validatorIndex) => {
       return assembleAttesterDuty(
         this.config,
         {publicKey: state.validators[validatorIndex].pubkey, index: validatorIndex},
         state,
         epoch
       );
     });
+    const subnets: Set<string> = new Set();
+    attesterDuties.forEach(duty => subnets.add(getCommitteeIndexSubnet(duty.committeeIndex)));
+    subnets.forEach(subnet => {
+      const peerIds = this.reps.getPeerIdsBySubnet(subnet);
+      if (peerIds.length < 3) {
+        // TODO: If an insufficient number of current peers are subscribed to the topic,
+        // the validator must discover new peers on this topic
```

Comment:
@wemeetagain how can I use discv5 to discover new peer given a subnet?

Reply:
So discv5 doesn't have topic discovery yet, but we can fetch the ENRs for all the peers we've found, and check the attnet value.
```typescript
discv5
  .kadValues()
  .filter((enr: ENR) => enr.get("attnet"))
  .filter((enr: ENR) => config.types.AttestationSubnets.deserialize(enr.get("attnet"))[subnet])
  .map((enr: ENR) => enr.peerId().then((peerId) => {PeerInfo.})

```

Reply:
If we already discovered them, we don't need to search it as they will send us stuff after we subscribe to topic?

Reply:
we may have discovered peers, but the libp2p machinery didn't dial them (because max number of peers reached)

Reply:
How do we decide what peers to disconnect to add new ones? :smile:

Reply:
I think libp2p would help figure that out.
They're planning on maintaining some sort of score for each peer in the peer store.
We would help feed the scores with data about success / fail network requests or good gossip.